File: ios/AccountsSDK.m
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(AccountsSDK, NSObject)

- (dispatch_queue_t)methodQueue
{
  return dispatch_get_main_queue();
}

+ (BOOL)requiresMainQueueSetup {
  return true;
}


RCT_EXTERN_METHOD(configureAccountsSDK: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(login: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(logout: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(logoutAll: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(refreshToken: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getMemberInfo: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getToken: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(isLoggedIn: (RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

@end

File: specs/AccountsSdk.ts
import type { TurboModule } from 'react-native';
import { TurboModuleRegistry } from 'react-native';

// TODO add the correct return types for getToken, getMemberInfo etc.
// In Android Accounts SDK module confirm all analytic key and value names are correct
export interface AccountsSdkSpec extends TurboModule {
  login(): Promise<void>;
  logout(): Promise<void>;
  logoutAll(): Promise<void>;
  isLoggedIn(): Promise<boolean>;
  getToken(): Promise<void>;
  getMemberInfo(): Promise<void>;
  refreshToken(): Promise<void>;
}

export default TurboModuleRegistry.getEnforcing<AccountsSdkSpec>('AccountsSdk');
